File: AccuPay/Domain/TimeEntry/TimeEntryPolicy.vb
Option Strict On

Public Class TimeEntryPolicy

    Private ReadOnly _settings As ListOfValueCollection

    Public Sub New(settings As ListOfValueCollection)
        _settings = settings
    End Sub

    Public ReadOnly Property LateHoursRoundingUp As Boolean
        Get
            Return _settings.GetBoolean("LateHours.RoundingUp")
        End Get
    End Property

    Public ReadOnly Property NightDiffBreakTime As Boolean
        Get
            Return _settings.GetBoolean("NightDiffPolicy.HasBreakTime")
        End Get
    End Property

    Public ReadOnly Property LateSkipCountRounding As Boolean
        Get
            Return _settings.GetBoolean("LateHours.SkipCountRounding")
        End Get
    End Property

    Public ReadOnly Property ComputeBreakTimeLate As Boolean
        Get
            Return _settings.GetBoolean("LateHours.ComputeBreakTimeLate")
        End Get
    End Property

    Public ReadOnly Property OvertimeSkipCountRounding As Boolean
        Get
            Return _settings.GetBoolean("OvertimeHours.SkipCountRounding")
        End Get
    End Property

    Public ReadOnly Property AbsencesOnHoliday As Boolean
        Get
            Return _settings.GetBoolean("Payroll Policy.holiday.allowabsence")
        End Get
    End Property

    Public ReadOnly Property RequiredToWorkLastDay As Boolean
        Get
            Return _settings.GetBoolean("Payroll Policy.HolidayLastWorkingDayOrAbsent")
        End Get
    End Property

    Public ReadOnly Property RequiredToWorkLastDayForHolidayPay As Boolean
        Get
            Return _settings.GetBoolean("HolidayPolicy.WorkLastDayForHolidayPay")
        End Get
    End Property

    Public ReadOnly Property HasNightBreaktime As Boolean
        Get
            Return _settings.GetBoolean("OvertimePolicy.NightBreaktime")
        End Get
    End Property

    Public ReadOnly Property RespectDefaultRestDay As Boolean
        Get
            Return _settings.GetBoolean("RestDayPolicy.RespectDefaultRestDay")
        End Get
    End Property

    Public ReadOnly Property IgnoreShiftOnRestDay As Boolean
        Get
            Return _settings.GetBoolean("RestDayPolicy.IgnoreShiftOnRestDay")
        End Get
    End Property

    Public ReadOnly Property RestDayInclusive As Boolean
        Get
            Return _settings.GetBoolean("Payroll Policy.restday.inclusiveofbasicpay")
        End Get
    End Property

End Class

File: Accupay.Data/Services/Policies/TimeEntryPolicy.cs
namespace AccuPay.Data.Services
{
    public class TimeEntryPolicy
    {
        private readonly ListOfValueCollection _settings;

        public TimeEntryPolicy(ListOfValueCollection settings) => _settings = settings;

        public bool LateHoursRoundingUp =>
                        _settings.GetBoolean("LateHours.RoundingUp");

        public bool NightDiffBreakTime =>
                        _settings.GetBoolean("NightDiffPolicy.HasBreakTime");

        public bool LateSkipCountRounding =>
                        _settings.GetBoolean("LateHours.SkipCountRounding");

        public bool ComputeBreakTimeLate =>
                        _settings.GetBoolean("LateHours.ComputeBreakTimeLate");

        public bool OvertimeSkipCountRounding =>
                        _settings.GetBoolean("OvertimeHours.SkipCountRounding");

        public bool AbsencesOnHoliday =>
                        _settings.GetBoolean("Payroll Policy.holiday.allowabsence");

        public bool RequiredToWorkLastDay =>
                        _settings.GetBoolean("Payroll Policy.HolidayLastWorkingDayOrAbsent");

        public bool RequiredToWorkLastDayForHolidayPay =>
                        _settings.GetBoolean("HolidayPolicy.WorkLastDayForHolidayPay");

        public bool HasNightBreaktime =>
                        _settings.GetBoolean("OvertimePolicy.NightBreaktime");

        public bool RespectDefaultRestDay =>
                        _settings.GetBoolean("RestDayPolicy.RespectDefaultRestDay");

        public bool IgnoreShiftOnRestDay =>
                        _settings.GetBoolean("RestDayPolicy.IgnoreShiftOnRestDay");

        public bool RestDayInclusive =>
                        _settings.GetBoolean("Payroll Policy.restday.inclusiveofbasicpay");

        public bool UseShiftSchedule =>
                        _settings.GetBoolean("ShiftPolicy.UseShiftSchedule");

        public bool ValidateLeaveBalance =>
                        _settings.GetBoolean("LeaveBalancePolicy.Validate");

        public bool PaidAsLongAsPresent =>
                        _settings.GetBoolean("TimeEntry.TimeLogAsPaymentRequirement");

        public bool PostLegalHolidayCheck =>
                        _settings.GetBoolean("HolidayPolicy.PostLegalHolidayCheck");
    }
}